src/libraries/System.Private.CoreLib/src/System/Threading/ThreadInt64PersistentCounter.cs

```diff
@@ -85,6 +93,9 @@ private sealed class ThreadLocalNode
             private uint _count;
             private readonly ThreadInt64PersistentCounter _counter;
 
+            internal ThreadLocalNode? _prev;
+            internal ThreadLocalNode? _next;
```

Comment:
```suggestion
            internal ThreadLocalNode _prev;
            internal ThreadLocalNode _next;
```
Maybe these can be non-nullable now to avoid `!` on all uses and better show the intent.

Reply:
You would need to add a dummy initialization to `this` to the constructor. It is actually a good thing since it avoids crash on the finalizer when we fail to add the node to the thread-local list.

Reply:
I have pushed a commit with this change.

Reply:
> add a dummy initialization to this to the constructor to make the compiler happy

I actually did it like this locally, but then it felt like trading assignments for nicer looking code without  `!/?` annotations.  I made is closer to original code in the end, but I think it is ok either way. It does look nicer without `!/?`.

Re: fiinalizer/Dispose. We are not very hardened for abuse here. 
We would not handle well if node is inserted twice, Disposed before finishing constructing, used after disposing, etc. I think it is not a big concern, since this is an internal class.